File: leetcode/1_two_sum.py
# https://leetcode.com/problems/two-sum/


class Solution:
    def twoSum(self, nums: list[int], target: int) -> list[int]:
        seen = {}

        for i, num in enumerate(nums):
            if target - num in seen:
                break
            seen[num] = i

        return [seen[target - num], i]

File: leetcode/1_two_sum.java
// https://leetcode.com/problems/two-sum/

package leetcode;

import java.util.HashMap;
import java.util.Map;

class Solution {
    public int[] twoSum(int[] nums, int target) {
        Map<Integer, Integer> seen = new HashMap<>();

        int i;
        for (i = 0; i < nums.length; i++) {
            if (seen.containsKey(target - nums[i])) {
                break;
            }
            seen.put(nums[i], i);
        }

        return new int[] {seen.get(target - nums[i]), i};
    }
}
